phases/repo.go

```diff
@@ -1,9 +1,104 @@
 package phases
 
-import "fmt"
+import (
+	"fmt"
+	"strings"
 
-// Repo ...
-func Repo() (string, error) {
-	fmt.Println("SetRepo")
-	return "", nil
+	"github.com/bitrise-io/go-utils/command"
+	"github.com/bitrise-io/go-utils/errorutil"
+	"github.com/bitrise-io/go-utils/log"
+)
+
+type urlParts struct{
+	host string
+	owner string
+	slug string
+}
+
+func parseURL(cloneURL string) urlParts {
+	parts := strings.SplitAfter(cloneURL, "https://")
+	if len(parts) > 1 {
+		// e.g. cloneURL=https://github.com/bitrise-io/go-utils.git
+		parts = strings.Split(parts[1], "/")
+		return urlParts{
+			host: parts[0],
+			owner: parts[1],
+			slug: parts[2],
+		}
+	} else {
+		// e.g. cloneURL=git@github.com:bitrise-io/go-utils.git
+		afterAt := strings.SplitAfter(parts[0], "git@")[1]
+		parts = strings.Split(afterAt, ":")
+		host := parts[0]
+		
+		afterHost := strings.SplitAfter(afterAt, ":")[1]
+		parts = strings.Split(afterHost, "/")
+		return urlParts{
+			host: host,
+			owner: parts[0],
+			slug: strings.TrimSuffix(parts[1], ".git"),
+		}
+	}
+}
+
+func buildURL(parts urlParts, protocol string) (cloneURL string) {
+	switch protocol {
+	case "https":
+		cloneURL = fmt.Sprintf("https://%s/%s/%s.git", parts.host, parts.owner, parts.slug)
+	case "ssh":
+		cloneURL = fmt.Sprintf("git@%s:%s/%s.git", parts.host, parts.owner, parts.slug)
+	}
+	return
+}
+
+func getProvider(cloneURL string) (string, error) {
```

Comment:
we have other option as well, not only these 3

Reply:
Indeed, however, If I return there, I have to put a dummy return at the end anyway, which could be confusing. I think the flow is clearer this way, but let us discuss offline.